src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -242,4 +248,58 @@ public Key createKey(Key options) throws Exception {
     public void deleteKey(String key) throws Exception {
         this.keysHandler.deleteKey(key);
     }
+
+    public String generateTenantToken(Map<String, Object> searchRules) throws MeiliSearchException {
+        return this.generateTenantToken(searchRules, new TenantTokenOptions());
+    }
+
+    /**
+     * Generate a tenant token
+     *
+     * @param searchRules A Map of string, object which contains the rules to be enforced at search
+     *     time for all or specific accessible indexes for the signing API Key.
+     * @param options A TenantTokenOptions, the following fileds are accepted: - apiKey: String
+     *     containing the API key parent of the token. If you leave it empty the client API Key will
+     *     be used. - expiresAt: A DateTime when the key will expire. Note that if an expiresAt
+     *     value is included it should be in UTC time.
+     * @return String containing the tenant token
+     * @throws MeiliSearchException if an error occurs
+     */
+    public String generateTenantToken(Map<String, Object> searchRules, TenantTokenOptions options)
+            throws MeiliSearchException {
+        // Validate all fields
+        if ((options.getApiKey() == null || options.getApiKey() == "")
+                && this.config.apiKey == "") {
+            throw new MeiliSearchException(
+                    "An api key is required in the client or should be passed as an argument.");
+        }
+        if (options.getExpiresAt() != null
+                && Date.from(Instant.now()).after((Date) options.getExpiresAt())) {
+            throw new MeiliSearchException("The date expiresAt should be in the future.");
+        }
+        if (searchRules == null) {
+            throw new MeiliSearchException(
+                    "The searchRules field is mandatory and should be defined.");
+        }
+
+        String secret;
+        if (options.getApiKey() == null) {
```

Comment:
@alallema if the config.apiKey is valid like `"88392che832ebh"` and the result of `options.getApiKey()` is `""` (empty string), your condition will not fall in the `MeiliSearchException` in the line 273. And will not fall in the 287 as well. Your result will be a `secret = "";`